draft-ietf-rtcweb-jsep.xml

```diff
@@ -2746,22 +2742,22 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
 
             <t>If codec preferences have been set for the associated
             transceiver, media formats MUST be generated in the
-            corresponding order, and MUST exclude any codecs not
-            present in the codec preferences or not present in the
-            offer. Note that non-JSEP endpoints are not subject to this
-            restriction, and might add media formats in the answer that
-            are not present in the offer, as specified in
-            <xref target="RFC3264"></xref>, Section 6.1. Therefore,
-            JSEP implementations MUST be prepared to receive such
-            answers.</t>
-
-            <t>Unless excluded by the above restrictions, the media
-            formats MUST include the mandatory audio/video codecs as
-            specified in
-            <xref target="RFC7874"></xref>, Section 3,
-            and
-            <xref target="RFC7742"></xref>, Section
-            5.</t>
+            corresponding order, regardless of what was offered,
+            and MUST exclude any codecs not present in the codec
+            preferences.</t>
+
+            <t>Otherwise, the media formats on the m= line
+            MUST be generated in the same order as those offered in the
+            current remote description, excluding any currently unsupported
+            formats. Any currently available media formats that are
+            not present in the current remote description MUST be added after
+            all existing formats.</t>
+
+            <t>In either case, the media formats in the answer MUST include
+            at least one format that is present in the offer, but MAY include
+            formats that are locally supported but not present in the offer,
+            as mentioned in <xref target="RFC3264" />, Section 6.1.
```

Comment:
If the answerer has an explicit preference, that wins, and so it would be C, A, B or C, B, A.

But if the preference is implicit, then the first one is correct (A, B, C) but not the second (which also must be A, B, C).